{
  try {
    listener.onComplete(this);
    return true;
  }
 catch (  Throwable th) {
    logger.warn("MessageFutureListener.onComplete() fail Caused:{}",th.getMessage(),th);
    return false;
  }
}
